fpptatweaks.php

```diff
@@ -489,26 +489,73 @@ function fpptatweaks_civicrm_postProcess($formName, $form) {
 function fpptatweaks_civicrm_searchColumns( $objectName, &$headers,  &$rows, &$selector ) {
   // Check if it is search contribution
   // !empty($rows) will prevent sql error if contact doesn't have contribution
-  if ($objectName == 'contribution' && !empty($rows)) {
+  if (
+    ($objectName == 'contribution'
+    || $objectName == 'event')
+    && !empty($rows)
+  ) {
+    // Since we don't want it to apply in event search, we will just return
+    if (CRM_Utils_System::getUrlPath() == 'civicrm/event/search'
+      || CRM_Utils_System::getUrlPath() == 'civicrm/contribute/search'
+    ) {
+      return;
+    }
+
     // Insert additional column header in the tab
     $insertedHeader = [
       'name' => E::ts('Inv #'),
       'field_name' => 'invoice_number',
       'direction' => CRM_Utils_Sort::DONTCARE,
       'weight' => 45,
     ];
-    $headers[] = $insertedHeader;
 
-    $contributionIds = CRM_Utils_Array::collect('contribution_id', $rows);
-    $keyPerContributionId = array_flip($contributionIds);
+    if ($objectName == 'event') {
+      // Column position of the inserted header, this is also useful
+      // ..when injecting the js for the value
+      $insertHeaderPosition = 5;
+      array_splice( $headers, $insertHeaderPosition, 0, [$insertedHeader] );
+      // Add weight on the header for the final position of the columns
+      foreach ($headers as $key => $head) {
+        $headers[$key]['weight'] = $key;
+      }
+
+      // Get participantIds array of column data
+      $participantIds = CRM_Utils_Array::collect('participant_id', $rows);
+      $contributionIds = [];
+      // Loop participantIds array to get contribution ID
+      foreach ($participantIds as $key => $participantId) {
+        $paymentDetails = civicrm_api3('ParticipantPayment', 'get', [
+          'sequential' => 1,
+          'participant_id' => $participantId,
+        ]);
+
+        // Save contributionIds as array
+        $contributionIds[$key] = $paymentDetails['values'][0]['contribution_id'];
+      }
+
+      // Flip for the rows position
+      $keyPerContributionId = array_flip($contributionIds);
+    }
+    else {
+      $headers[] = $insertedHeader;
+      $contributionIds = CRM_Utils_Array::collect('contribution_id', $rows);
+      $keyPerContributionId = array_flip($contributionIds);
+    }
 
     $contributions = \Civi\Api4\Contribution::get()
       ->addSelect('id', 'invoice_number')
       ->addWhere('id', 'IN', $contributionIds)
       ->execute();
 
-    foreach ($contributions as $contribution) {
+    foreach ($contributions as $key => $contribution) {
       $rows[$keyPerContributionId[$contribution['id']]]['invoice_number'] = $contribution['invoice_number'];
     }
+
+    if ($objectName == 'event') {
```

Comment:
I don't think we need a separate `if ($objectName == 'event') {` block here, and instead we could just put this code into the `if ($objectName == 'event') {` that begins on [line 512](https://github.com/twomice/com.joineryhq.fpptatweaks/pull/6/files#diff-87ac059cb9fbcc07df9cfb9e715a5cea8906015df4bc8b635d0ad6d0c1aa0ae5R512). If you agree, please make that change.

Reply:
@twomice I think it is needed since we will need to get the updated row first and inject the var eventRows and js. I moved the columnPosition addVars at the top though.

Reply:
I see, you're modifying `$rows` in line 552, and then passing that modified array to `addVars` on line 577. That explains the need for this separate `if` block.

In this case, you might as well leave the `... ->addVars('columnPosition')` line here with the other `CRM_Core_Resources::singleton()->...` lines. Please go ahead and move it back here @lucky091588